shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,84 @@
+import json
+import math
+import nibabel as nib
+import os
+
+
+def read_nii(niiPath):
+    """
+
+    Args:
+        niiPath (str): Path to the nifti file
+
+    Returns:
+        image:
+        info:
+        jsonData:
+    """
+
+    info = nib.load(niiPath)
+    image = info.get_fdata()
```

Comment:
I'm skeptical about nibabel's design here. [the docs](https://nipy.org/nibabel/reference/nibabel.dataobj_images.html#nibabel.dataobj_images.DataobjImage.get_fdata) show that this is up-casting everything to 64-bit floats. I think that's very expensive, potentially, if the data on disk isn't in that format because it means every element gets copied, converted, and stored in RAM.

https://nipy.org/nibabel/reference/nibabel.dataobj_images.html#nibabel.dataobj_images.DataobjImage.get_data explains

> If you don’t care about the predictability of the return data type, and you want the minimum possible data size in memory, you can replicate the array that would be returned by img.get_data() by using np.asanyarray(img.dataobj).

I think that sounds better. It means the datatype is determined by the file, and the read-cache is efficient.

```suggestion
    image = np.asanyarray(info.dataobj)
```